src/WooCommerce.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/**
+ * @file
+ * Contains \Netzstrategen\Gallerya\WooCommerce.
+ */
+
+namespace Netzstrategen\Gallerya;
+
+/**
+ * WooCommerce custom functionality.
+ */
+class WooCommerce {
+
+  /**
+   * Adds srcset attributes to image links to make them reponsive on lightgallery.
+   *
+   * @see https://sachinchoolur.github.io/lightGallery/demos/responsive.html
+   *
+   * @implements woocommerce_single_product_image_thumbnail_html
+   */
+  public static function woocommerce_single_product_image_thumbnail_html($html, $thumbnail_id) {
+    $srcset = wp_get_attachment_image_srcset($thumbnail_id, 'shop_single');
+    return preg_replace('/(<a\s+)/i', '<a data-srcset="' . $srcset . '" ', $html);
```

Comment:
It looks like all you want to do is to move the `<img srcset>` attribute into a wrapping `<a data-srcset>` — and I assume also remove the `srcset` attribute from `img` as the operation wouldn't make sense otherwise.

90% of this plugin is needs JavaScript, so why aren't you doing this in JavaScript?

Reply:
1. The image used for ligthGallery have not `srcset`, so we can not "move" nothing.
2. The reason why there is not `srcset` is because it should be come from the parent `<a data-srcset>`. Is the only place we could specify it.

To better describe it, look at the following images.

This is the regular product image were we're outputting the `srcset` attribute.
<img width="740" alt="schermata 2017-11-29 alle 09 57 53" src="https://user-images.githubusercontent.com/2393718/33366538-ccf35fea-d4eb-11e7-9931-226f29633adc.png">

This is the same product image used by ligthGallery. As you can see no `srcset` here, anywhere.
<img width="872" alt="schermata 2017-11-29 alle 09 57 39" src="https://user-images.githubusercontent.com/2393718/33366586-f764f1da-d4eb-11e7-9289-dc14a8f6d342.png">

Reply:
In my opinion the solution provided by Luca fits well. But if it's easily possible to get image information via Javascript, the JS solution would be bit easier as you can simply call `$(element).data('srcset', values)`. Maybe have a look at [this](https://wordpress.stackexchange.com/a/237574/77421) SO post

Reply:
Yep, IMHO we should not use JS here: too complex. With JS we should:

- Check the WC product image that correspond to the one included in the `lg-img-wrap`.
- Extract the `src-set` from it and apply it to the `<a>` parent of the lightGallery image.

Sounds really more tricky then this 2 PHP lines, especially the first point were, to compare the related images we should check the `src`.

Sincerely, this PHP solution works, works fine and seems not so dirty.

Reply:
Why is there no srcset on the image?  Shouldn't there be one, regardless of whether any lightbox is used or not?

As far as I know we don't need a template override for adding srcset to images, because there's a filter hook for image attributes.

If there was a srcset on the image element, then that attribute could be easily moved with JS to the wrapping link element.  That in turn would avoid the template override.

Reply:
Adding a `srcset` on that image is exactly what we're trying to do it here and the only way to have a `scrset` on that image is adding a `data-srcset` (or the other data attr suggested by the plugin) to the wrapping `a` element.

The filter you are talking about are the hooking the product image and the product thumbnails. But we don't need this. We are already using `srcset` for the product images and thumbnails.

What we need is use `srcset` for the lightGallery image, the one is using `lg-image` class. This element is directly created by JS, not by PHP: no hooks to control this. The only way to add a `scrset` to this element is the one I used.

Reply:
Ok, sorry — I get only now what you mean.

But, really, I can not understand why a JS solution should be better then the PHP one proposed by me. It's just a few line PHP filter, with no template override. Could you explain me the advantage of do so?

Anyway we should copy, not move the `srcset` because we still need it on the regular product images/thumbnails.

Reply:
> Anyway we should copy, not move the srcset because we still need it on the regular product images/thumbnails.

Yes, exactly, we can copy the `srcset` attribute from each image into a `data-srcset` attribute on each link.

The most efficient way to do this is in JavaScript, right before initializing the gallery/slider/lightbox.  JavaScript is the appropriate technology, because neither the backend nor the CSS/appearance needs the information.

Alternatively, instead of changing anything in the DOM, you can also consider to use lightGallery's dynamic mode and pass the necessary data directly after reading it from different sources: https://sachinchoolur.github.io/lightGallery/demos/dynamic.html